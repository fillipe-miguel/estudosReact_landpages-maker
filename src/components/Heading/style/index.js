import styled from 'styled-components';

// const titleSize = {
//   small: (theme) => theme.fonts.sizes.small,
//   medium: (theme) => theme.fonts.sizes.medium,
//   large: (theme) => theme.fonts.sizes.large,
// };

export const Container = styled.h1`
  color: ${({ theme, colorDarker }) =>
    colorDarker ? theme.colors.black : theme.colors.white};

  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
  font-size: ${({ theme, size }) => theme.fonts.sizes[size]};

  @media ${({ theme }) => theme.media.lteMedium} {
    font-size: ${({ theme, size }) =>
      size === 'xxxlarge'
        ? theme.fonts.sizes.xlarge
        : theme.fonts.sizes.xxxlarge};
  }
`;

export default Container;

// font-size: ${({ theme, size }) => titleSize[size](theme)};
