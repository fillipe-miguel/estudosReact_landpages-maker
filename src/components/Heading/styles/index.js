import styled, { css } from 'styled-components';

export const HeadingRoot = styled.h1`
  // ================================= Para pegar as props =================================
  ${({ theme, ...props }) => css`
    color: ${props.colorDarker
      ? theme.colors.primaryColor
      : theme.colors.white};

    text-transform: ${props.uppercase ? 'uppercase' : 'none'};

    font-size: ${theme.fonts.sizes[props.size]};
  `}

  // ================================= Medias =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      font-size: ${props.size === 'xxxlarge'
        ? theme.fonts.sizes.xlarge
        : theme.fonts.sizes[props.size]};
    }
  `}
`;

export default HeadingRoot;

// ================================= Antiga organização =================================

// font-size: ${({ theme, size }) => titleSize[size](theme)};

// @media ${({ theme }) => theme.media.lteMedium} {
//   font-size: ${({ theme, size }) =>
//     size === 'xxxlarge'
//       ? theme.fonts.sizes.xlarge
//       : theme.fonts.sizes.xxxlarge};
// }

// color: ${({ theme, colorDarker }) =>
//     colorDarker ? theme.colors.black : theme.colors.white};

// text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
// font-size: ${({ theme, size }) => theme.fonts.sizes[size]};
