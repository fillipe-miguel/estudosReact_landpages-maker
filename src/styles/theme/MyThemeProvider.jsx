import P from 'prop-types';
import React from 'react';
// Styles
import { ThemeProvider } from 'styled-components';
import GlobalStyles from '../global-styles';
import theme from './theme';

function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
}

MyThemeProvider.propTypes = {
  children: P.node.isRequired,
};

export default MyThemeProvider;
