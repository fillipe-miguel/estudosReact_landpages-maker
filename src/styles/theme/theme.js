const theme = {
  // Your theme HERE
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#DC143C',

    lightBg: '#FFF',
    darkBg: '#0A1128',

    white: '#FFF',
    black: '#000',
  },
  fonts: {
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      // Pulei um pois achei que n√£o usaria !
      xxxlarge: '6.4rem',
    },
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4rem',
    xxlarge: '4.8rem',
  },
  medias: {
    lteMedium: '(max-width: 768px)',
  },
};

export default theme;
