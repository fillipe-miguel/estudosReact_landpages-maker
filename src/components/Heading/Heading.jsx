import P from 'prop-types';
import React from 'react';
import * as Styled from './style';

function Heading({ children, colorDarker, level, size, uppercase }) {
  return (
    <Styled.Container
      colorDarker={colorDarker}
      as={`h${level}`}
      size={size}
      uppercase={uppercase}
    >
      {children}
    </Styled.Container>
  );
}

Heading.defaultProps = {
  size: 'large',
  uppercase: false,
  colorDarker: true,
};

Heading.propTypes = {
  children: P.string.isRequired,
  colorDarker: P.bool,
  level: P.oneOf(['1', '2', '3', '4', '5', '6']).isRequired,
  size: P.oneOf(['medium', 'large', 'xlarge', 'xxlarge', 'xxxlarge']),
  uppercase: P.bool,
};

export default Heading;
