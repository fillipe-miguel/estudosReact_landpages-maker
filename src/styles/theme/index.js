import MyThemeProvider from './MyThemeProvider';
import theme from './theme';

export { MyThemeProvider, theme };
