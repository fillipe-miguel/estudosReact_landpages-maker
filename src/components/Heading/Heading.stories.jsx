/* eslint-disable react/function-component-definition */ // ISSO Ã‰ PARA O ESLINT DEIXAR EU USAR ARROW FUNCTION
import Heading from '.';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    children: 'Meu titulo funcionando!!',
    level: '1',
    uppercase: false,
  },
  argsTypes: {
    children: { type: 'string' },
    colorDarker: { type: 'boolean' },
    uppercase: { type: 'boolean' },
  },
};

// ===================== LIGHT TEST
export const InLightBG = (args) => <Heading {...args} />;

InLightBG.args = {
  colorDarker: true,
};

InLightBG.parameters = {
  backgrounds: {
    default: 'light',
  },
};

// ===================== DARK TEST
export const InDarkBG = (args) => <Heading {...args} />;

InDarkBG.args = {
  colorDarker: false,
};

InDarkBG.parameters = {
  backgrounds: {
    default: 'dark',
  },
};
