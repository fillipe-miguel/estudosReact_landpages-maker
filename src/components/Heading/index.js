import Heading from './Heading';
import * as HeadingStyles from './styles';

export { HeadingStyles };
export default Heading;
