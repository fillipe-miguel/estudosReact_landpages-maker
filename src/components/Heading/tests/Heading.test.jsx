/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
// Styles
import MyThemeProvider from '../../../styles/theme/MyThemeProvider';
import theme from '../../../styles/theme/theme';
// Components
import Heading from '..';

describe('<Heading />', () => {
  // ================================= test 1 =================================
  it('Should render default', () => {
    render(
      <MyThemeProvider>
        <Heading level="1">texto</Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    // Cor padrão | FontSize padrão
    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.large,
      'text-transform': 'none',
    });
  });

  // ================================= test 2 =================================
  it('Should render with white color', () => {
    render(
      <MyThemeProvider>
        <Heading level="1" colorDarker={false}>
          texto
        </Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    // Cor white
    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  // ================================= test 3 =================================
  it('Should render with xxxlarge size', () => {
    render(
      <MyThemeProvider>
        <Heading level="1" size="xxxlarge">
          texto
        </Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    // FontSize grandão
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.xxxlarge,
    });
  });

  // ================================= test 4 =================================
  it('Should render with large size', () => {
    render(
      <MyThemeProvider>
        <Heading level="1" size="large">
          texto
        </Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.large,
    });
  });

  // ================================= test 5 =================================
  it('Should render correct font-size when screen mobile', () => {
    render(
      <MyThemeProvider>
        <Heading level="1" size="xxxlarge">
          texto
        </Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    // Mudando a fonte quando a tela fica pequena
    expect(heading).toHaveStyleRule('font-size', theme.fonts.sizes.xlarge, {
      media: theme.medias.lteMedium,
    });
  });

  // ================================= test 6 =================================
  it('Should render with uppercase', () => {
    render(
      <MyThemeProvider>
        <Heading level="4" uppercase>
          texto
        </Heading>
      </MyThemeProvider>,
    );
    // Heading ==> h1 | h6 // ! Pegando o componente renderizado pelo texto
    const heading = screen.getByRole('heading', { name: 'texto' });

    // ! Checando estilo
    // Text transform
    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  // ================================= Snapshot =================================
  it('Should match snapshot', () => {
    const {container} = render(
      <MyThemeProvider>
        <Heading level="1">SNAP</Heading>
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
