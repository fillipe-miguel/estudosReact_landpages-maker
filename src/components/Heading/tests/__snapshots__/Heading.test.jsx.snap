// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  text-transform: none;
  margin-bottom: 3.2rem;
  font-size: 3.2rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 3.2rem;
  }
}

<h1
  class="c0"
>
  SNAP
</h1>
`;
